### DISCUSSION TITLE
DirecTV Remotes RC73x (Genie) / RC82v (Gemini)

### DISCUSSION THREAD

**Discussion by skybasemonkeyha** [PRProd/HA-Firemote#392] [New Device Family Support Request]
### Feature Request

Request DirecTV Remote RC73 (Genie) and RC82 (Gemini/Osprey)
Hi there for the directv remotes the command available in the DirecTV Remote Integration so it would be similar to the AndroidTV Remote Integration commands mostly the same but some are undefined. 

buttons for the rc73 listed top to button left to right and its associated directv integration remote command
power on = media_player.turn_on (remote doesn't seem to work right as the integration makes the entity when off, unavailable)
power off = remote.turn_off
guide = remote.send_command | guide
menu = remote.send_command | menu
list = remote.send_command | list
record = remote.send_command | record
dpad_up = remote.send_command | up
exit = remote.send_command | exit
dpad_left = remote.send_command | left
center = remote.send_command | select
dpad_right = remote.send_command | right
red_circle (delete) = remote.send_command | red
dpad_down = remote.send_command | down
info = remote.send_command | info
mute = tv related button
volume_up = tv related button
volume_down = tv related button
chan_up =  remote.send_command | chanup
chan_down =  remote.send_command | chandown
previous =  remote.send_command | prev
replay =  remote.send_command | replay
advance =  remote.send_command | advance
rewind =  remote.send_command | rew
play/pause =  remote.send_command | play
forward =  remote.send_command | ffwd
number_1 =  remote.send_command | 1
number_2 =  remote.send_command | 2
number_3 =  remote.send_command | 3
number_4 =  remote.send_command | 4
number_5 =  remote.send_command | 5
number_6 =  remote.send_command | 6
number_7 =  remote.send_command | 7
number_8 =  remote.send_command | 8
number_9 =  remote.send_command | 9
dash =  remote.send_command | dash
number_0 =  remote.send_command | 0
enter =  remote.send_command | enter

every command that matches a command for the genie (rc73) you can use the genie command as well.
but here are the adb commands commands. I have a gemini connected to a genie so my genie remote works with the gemini in ir mode all buttons work airgo the directv integration commands still work. 

missing buttons on the rc82 that were on the rc73
red circle (there is one but its record now)
advance 
replay
menu (replaced by home)
info (replaced by options)

new buttons on rc82 not on the rc73 the remote 
home (replaces menu)
options (replaces info) 
input 
apps

who ever develops the rc82 remote i recommend you use the directv integration commands as much as possible,
but here are the as many adb commands i discovered as well.
power on = android.adb_command | input keyevent 26
power off = android.adb_command | input keyevent 223
guide = androidtv.adb_command | input keyevent 172
home = androidtv.adb_command | input keyevent 3
list = androidtv.adb_command | input keyevent 173
record = androidtv.adb_command | input keyevent 130
dpad_up = androidtv.adb_command | input keyevent 19
exit = androidtv.adb_command | input keyevent 135
dpad_left = androidtv.adb_command | input keyevent 21
center = androidtv.adb_command | input keyevent 23
dpad_right = androidtv.adb_command | input keyevent 22
back = androidtv.adb_command | input keyevent 4
dpad_down = androidtv.adb_command | input keyevent 20
info = androidtv.adb_command | input keyevent 165
volume_up = tv related button
volume_down = tv related button
chan_up = androidtv.adb_command | input keyevent 166
chan_down =  androidtv.adb_command | input keyevent 167
mute = tv related button
google assistant = androidtv.adb_command | input keyevent 219
previous =  androidtv.adb_command | input keyevent 229
rewind =  androidtv.adb_command | input keyevent 89
play/pause =  androidtv.adb_command | input keyevent 85
forward = androidtv.adb_command | input keyevent 90
number_1 =  androidtv.adb_command | input keyevent 8
number_2 =  androidtv.adb_command | input keyevent 9
number_3 =  androidtv.adb_command | input keyevent 10
number_4 =  androidtv.adb_command | input keyevent 11
number_5 =  androidtv.adb_command | input keyevent 12
number_6 =  androidtv.adb_command | input keyevent 13
number_7 =  androidtv.adb_command | input keyevent 14
number_8 =  androidtv.adb_command | input keyevent 15
number_9 =  androidtv.adb_command | input keyevent 16
number_0 =  androidtv.adb_command | input keyevent 7
apps =  androidtv.adb_command | input keyevent 136

adb commands for gemini that are not on the rc82 remote but on the rc73 
menu = androidtv.adb_command | input keyevent 82
advance = androidtv.adb_command | input keyevent 274

no working adb commands for
replay
red_circle
dash

Genie Receivers proven to work
in order to use the Genie Mini Receivers (C41, C41K, C41W , C41KW, C51, C51K, C51W , C51KW,  C61, C61K, C61W , C61KW) you Need to Have an HR44 (Genie 1) or HR54 (Genie 1) or HS17 (Genie 2) integration developers say this may work on older units like the H25 and the HR34 never had the H25 when using home assistant so couldn't tell you also never had an HR34
In testing I Have used the HS17 with a C51, C61 and C61K as well as the C71KXW (Gemini) 

Using all three integrations (AndroidTV Remote, Android Debuging Bridge, and DirecTV) i have been able to get all available features of the remote and the receivers to work accordingly. I have created my own template, but i'd like some one to create one like they did with the firetv remote and now the rokus.

**Comment by PRProd** [comment]
Greetings @skybasemonkeyha!

THANK YOU for all of the effort you put into making such a detailed request!  Simply amazing!  I fully understand exactly how much work it takes to do all of that, so you've definitely earned my respect!  :+1: 

To answer your "who ever makes the..." question, it's just me.  There are two other one-time contributors to this project, but their contributions were simply app launcher button definitions (which I appreciate!), not full UI elements like what you are requesting.

So, here's the potentially bad news... I have a system in place to decide what gets added next.  There is a discussion topic named [Poll: Vote for what gets added next! #41](https://github.com/PRProd/HA-Firemote/discussions/41) where I gauge user interest.  So far, the Firemote project's direction has been 100% directed by the poll results from that topic.  Additionally, I have a personal rule that I will only add support for hardware that I own.  (yes, I literally own a stack of media players now)  I did make one small compromise to that rule with issue #303, and because the requester of that issue was willing to work with me so closely, we were able to add that to the project.  It was simply an Amazon Fire product that I didn't own, not a brand new family of devices, so most of the foundation was already there to begin with.

This is what we need to do first:
 - I am going to move this issue to a discussion to stay in line with the project's organizational arrangements
 - I would like for you to cast your vote in discussion #41 for this request

I know that you are savvy enough to have compiled such a detailed request.  If I decide to work on this outside of the normal voting, I have some questions before we move further:
 - How much time are you willing or able to contribute to working with me on this?
 - Are you willing to do remote (remote) sessions with me if we go down this path since I do not own a DirecTV system?
 - Do you know anybody else who uses DirecTV and Home Assistant who might be willing to do some testing after a beta is built?
 - How willing are you to answer questions and possibly provide some support after DirecTV is added?

Again, THANK YOU!  I'm excited to hear back.

-Doug

**Reply by skybasemonkeyha** [reply]
Doug all I need you to do is map out the remotes, i can handle the rest, I dont have as much time to help as i am a doctor and a pilot and engineer but i can help out when i can but to make your life easier if you choose to do it 
![604a0e2e-a27e-47f3-aa85-de3a59ea6f8f-0](https://github.com/PRProd/HA-Firemote/assets/131741634/1b9fbd9a-78d5-4369-8cb0-3a40206d7740)

as you can see i mapped mine out just like the remote but i didn't put it against the actual image. i can provide you with my map if you like

as you can also see i have a vizio xrt140 mapped out, however i have yet to learn all the commands so it is mapped out partially using an infrared wifi remote unit. I have one for a samsung tizen, Insignia tv, Sony Bravia, mapped out where i can with commands, rest of it was wifi remote. all of them laid out like the original remote.

My Wi-Fi remotes are different brands of the Tuya Wi-Fi Remotes,  Connected and Manually Mapped out using the Smart Life App
Linked to Developer, Imported into HOOBS / Homebridge, Bridged to Home Assistant with HomeKit.

I have yet to make a remote for my LG Portable Air Conditioner, LG TV, RCA TV, I did have one for my Hi-Sense TV but since it was down for a while now it was taken off my home assistant, 

My Hi-Sense TV I could never properly link to OpenHAB, HomeBridge, or Home Assistant it is the only SmartTV I have yet to connect.

I recent finally figured out the homekit issue i was having with my insignia firetv, by using Keep Alive and newer home assistants. I need Home Kit for Source Input for them as adb doesn't support it, and I am going today work around jeffs Current App Issue problem caused by the config flow, by using adb on openHAB,

My Complex Home is controlled by 3 Home Assistant Servers,  2 OpenHAB Servers and HOOBs Server, If all goes well with the rewriting of my android tv scripts, and entities and virtual devices Home assistant server #2 will be discontinued, as it was specifically setup for the insignia/toshiba firetv issue. 

I have combined multiple integrations on multiple platforms and created templated receivers, air conditioners, and televisions.

My most recent problem solved started about 2 years ago, when i got my viper smart start. I initially had it through smart things but they stopped it in january 2023 for a year now i had been trying to get it working other ways, I was able to create the commands to be sent using xampp and rest_commands and a postman script but apparently the postman script was flawed when i came to status updates, for a year now i have been retrying to get status updates with the smart start, trying with home assistant and hoobs and even with openhab, it was just yesterday though i saw it many times over, and didn't think i can fix it in that postman script because i just keep overlooking the html pre header which was stopping the json path transformations. yesterday for the first time in almost 2 years i finally have status updates with my viper smart start, Including a low battery alert because my battery was replaced with out my consent with a cheap battery at pepboys and unfortunately has caused starting problems. It is always the simple things we over look.

**Comment by PRProd** [comment]
Are these the physical remotes that you use?
 - https://a.co/d/8CL7hur
 - https://a.co/d/62zk049

Do you own any spares that you'd be willing to send me?

**Reply by skybasemonkeyha** [reply]
yes those are the right remotes but as far as a spare rc82 i am sorry i don't have a spare, they are newer and i dont have any spares yet The Gemini Receivers were only released 12 months earlier. I'd be happy to send you one of my rc73s though

**Reply by skybasemonkeyha** [reply]
the idea for this is to get the picture mapped out to linked command namespaces just like what you have done for the other android. if you would let me into your process i can probably map it out for ya. maybe get it for some universal remotes as well if you can get me into your process. I just mapped out the vizio remote, to virtual non-ir commands. using openhab as its command parser. still having issues however with the app buttons.

**Reply by PRProd** [reply]
The "assigning a command to a button" mapping isn't really the first priority.  It's also possible that you didn't realize that the remotes are not actual pictures.  A lot of people who do this sort of thing will use pictures and just overlay a "button target mapping", but that method limits the amount of features that can be added to the control surface.

There are no pictures on the base remotes.  SVG images are used for logos, but everything else is created through HTML and CSS.  That's why I'm interested in having a remote in my hands.  I can't meticulously draw something from just photos alone.

**Reply by PRProd** [reply]
If you are interested in "getting into the process", the whole source code is open.

Remote Rendering HTML code: https://github.com/PRProd/HA-Firemote/blob/main/dist/HA-Firemote.js#L2936-L4560
Remote Styling CSS code: https://github.com/PRProd/HA-Firemote/blob/main/dist/HA-Firemote.js#L223-L2511

**Reply by skybasemonkeyha** [reply]
i didn't realize you are drawing them, pm me your info i'll send you out both remotes in a few days, i'm flying the medivac for the first time in 2 months since i broke my angle and sprained my knee then i got a shift at the hospital afterwards, I will not be available tomorrow.

**Reply by skybasemonkeyha** [reply]
if you want the one for the bravia and the bravia integration as well i got a busted one and a samsung one for the samsung integrations

**Comment by PRProd** [comment]
@skybasemonkeyha ~

I'm still quite hesitant to bypass the voting process (possibly upsetting everyone participating in that [poll](https://github.com/PRProd/HA-Firemote/discussions/41)) to add something that I don't own myself.  That's really two big reasons that go against my unofficial rules.

Even though I might get some help with initial beta testing, and follow-up technical support from you and possibly @thos-edison, I don't know if I'd ever feel 100% confident that I've worked out all of the bugs before publishing something.  This project is something that I've worked very hard on, so its integrity is important to me.  I have a much better chance of doing it right the first time if I can test things thoroughly in my own lab.

There have also been a lot of people voting and waiting for results in [the poll](https://github.com/PRProd/HA-Firemote/discussions/41).  While I do mention that buying or sending me a device will likely sway me in choosing said device, you and I have only discussed mailing the physical remotes.  That's only part of what I would need.

This could all change, though.  If I have a local friend who uses these devices AND Home Assistant, they might be willing to put up with me for a few weekends while I work on this.  But as of right now, I think it's smarter to stick with the plan that's in place.

I recommend voting for **Other** in the [poll](https://github.com/PRProd/HA-Firemote/discussions/41) and then adding your request details in [the proper thread](https://github.com/PRProd/HA-Firemote/discussions/41#discussioncomment-5293057) just the same as everyone else.  It's looking like **Homatics** is going to be the next winner, and I don't own any of those devices yet either.

Sorry to potentially disappoint you on this decision for now, I hope you understand.

-Doug

**Reply by skybasemonkeyha** [reply]
then don't make me a priority then be fair to others

**Reply by skybasemonkeyha** [reply]
i see the samsung is on your list, if you want i can send you some samsung remotes oh btw toshiba and samsung are the same company and they also own insignia
